go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -2120,6 +2123,261 @@ func (s *VtctldServer) UpdateCellsAlias(ctx context.Context, req *vtctldatapb.Up
 	}, nil
 }
 
+// ValidateSchemaKeyspace is part of the vtctlservicepb.VtctldServer interface
+func (s *VtctldServer) ValidateSchemaKeyspace(ctx context.Context, req *vtctldatapb.ValidateSchemaKeyspaceRequest) (*vtctldatapb.ValidateSchemaKeyspaceResponse, error) {
+	shards, err := s.ts.GetShardNames(ctx, req.Keyspace)
+	if err != nil {
+		return nil, vterrors.Wrapf(err, "GetShardNames(%v) failed: %v", req.Keyspace, err)
+	}
+
+	sort.Strings(shards)
+	if len(shards) == 1 {
+		validateSchemaShardReq := &vtctldatapb.ValidateSchemaShardRequest{
+			Keyspace:       req.Keyspace,
+			ExcludeTables:  req.ExcludeTables,
+			Shard:          shards[0],
+			IncludeVSchema: req.IncludeVSchema,
+			IncludeViews:   req.IncludeViews,
+		}
+
+		if _, err := s.ValidateSchemaShard(ctx, validateSchemaShardReq); err != nil {
+			return nil, vterrors.Wrapf(err, "ValidateSchemaShard(%+v) failed: %v", validateSchemaShardReq, err)
+		}
+		return &vtctldatapb.ValidateSchemaKeyspaceResponse{}, nil
+	}
+
+	var (
```

Comment:
this is minor, but stylistically, most of the other functions put the concurrency variables first (waitgroup, then error recorder). let's keep it consistent